### DISCUSSION TITLE
Issues Running GitHub Actions Locally with act on Apple M1 Chip

### DISCUSSION THREAD

**Discussion by nh916** [nektos/act#2157] [Q&A]
Hello everyone,

I'm encountering a problem when trying to run GitHub Actions locally using `act` on my MacBook with an Apple M-series chip. I have Docker installed and am trying to run the workflow with the following command:

```bash
act -j mypy-check --container-architecture linux/amd64
```

However, the process seems to hang indefinitely, with the last output being related to Docker pulling an image:

```
[MyPy Type Check/MyPy Type Checking] 🚀 Start image=catthehacker/ubuntu:act-latest platform=linux/amd64 username= forcePull=true
[MyPy Type Check/MyPy Type Checking]   🐳 docker pull image=catthehacker/ubuntu:act-latest platform=linux/amd64 username= forcePull=true
[MyPy Type Check/MyPy Type Checking] Using DockerAuthConfig authentication for Docker pull
```

Here are the steps I've taken so far:
- Ensured Docker is running and can pull images on my machine.
- Checked my network connection, which is stable and without any issues.
- Logged into the Docker desktop client

Is there any additional configuration I need to do for `act` to work with Docker on an Apple M-series chip? Or are there any known issues with Docker image compatibility?

I would appreciate any insights or suggestions on how to resolve this hang-up.

Thank you!

**Comment by madisonleavo** [comment]
I am running on M1 with no issues. might be worth pulling that image manually? it is pretty large.
and once I have the image, I do set `--pull=false` 

i'm not sure what the `-j mypy-check` is doing. you can try running act with `-v` to see if there are any more clues.

% act --version
act version 0.2.59

% docker --version
Docker version 24.0.7, build afdd53b
